working/augmentation-libraries/feature-specification.md

```diff
@@ -163,78 +182,72 @@ Augmentation declarations include:
-
-An augmentation that replaces the body of a function, may also want to preserve
-and run the code of the augmented declaration (hence the name "augmentation").
-It may want to run its own code before the augmented code, after it, or both.
-To support that, we allow a new expression syntax inside the “bodies” of
-augmenting declarations (function bodies, constructor bodies, and variable
-initializers). Inside an expression of a member marked `augment`, the identifier
-`augmented` can be used to refer to the augmented function, getter, or setter
-body, or variable initializer. This is a contextual keyword within `augment`
-declarations, and has no special meaning outside of that context. See the next
-section for a full specification of what `augmented` means, and how it must be
-used, in the various contexts.
-
-**Note:** Within an `augment` member, a reference to a member by the same name
-refers to the final version of the member (and not the one being augmented). The
-only way to reference the augmented member is through the `augmented`
-expression.
+A non-local variable induces a getter and possibly a setter. It is possible
+to augment getters and setters (this is a kind of function augmentation),
+including the ones that are induced by variables.
+
+*This means that augmentation of a variable declaration can be a getter
+declaration that augments the induced getter, a setter declaration that
+augments the induced setter, or a variable declaration that augments the
+initializer. The augmenting declarations can themselves be an implicitly
+induced getter or setter, or an explicitly declared one.*
+
+*Note that an abstract variable declaration and an external variable
+declaration correspond to a getter declaration and possibly a setter
+declaration. There is no notion of an initializing expression, and hence
+also no support for augmenting it.*
+
+These operations cannot be expressed today using only imports, exports, or
+part files. Any Dart file (library file or part file) can contain
+augmentation declarations.
+
+A type augmentation can add new members to an existing type, or augment a
+member declaration in the same context *(that is, in the same type
+augmentation, or in a type declaration that it augments)*.
+
+Because of augmentations, non-abstract class, mixin, mixin class, enum,
+extension type, and extension declarations are now allowed to contain
+abstract member declarations, as long as those members are equipped with a
+body by an augmentation declaration. _This is primarily useful for macros,
+which may be used to provide a body for an abstract member._
+
+An augmentation that replaces the body of a function may also want to
+preserve and run the code of the augmented declaration (hence the name
+"augmentation").  It may want to run its own code before the augmented
+code, after it, or both.  To support that, we allow a new expression syntax
+inside the “bodies” of augmenting declarations (function bodies,
```

Comment:
Tiny nit: I generally try to avoid smart quotes (`“` and `”`) in Markdown files and keep things as plain ASCII (`"`). Whether I do that for any logical reason versus just being weird about arbitrary consistency is an open question. :)